src/librustc_error_codes/error_codes/E0687.md

```diff
@@ -0,0 +1,50 @@
+in-band lifetimes cannot be used in `fn`/`Fn` syntax
+
+Erroneous code examples:
+
+```compile_fail,E0687
+#![feature(in_band_lifetimes)]
+
+fn foo(x: fn(&'a u32)) {} // error!
+
+fn bar(x: &Fn(&'a u32)) {} // error!
+
+fn baz(x: fn(&'a u32), y: &'a u32) {} // error!
+
+struct Foo<'a> { x: &'a u32 }
+
+impl Foo<'a> {
+    fn bar(&self, x: fn(&'a u32)) {} // error!
+}
+```
+
+```compile_fail,E0687
+#![feature(in_band_lifetimes)]
+
+fn bar<F>(x: &F) where F: Fn(&'a u32) {} // error!
+
+fn baz(x: &impl Fn(&'a u32)) {} // error!
+```
+
+lifetimes used in `fn` or `Fn` syntax must be explicitly
+declared using `<...>` binders. For example
+
+```
+fn foo<'a>(x: fn(&'a u32)) {} // ok!
+
+fn bar<'a>(x: &Fn(&'a u32)) {} // ok!
+
+fn baz<'a>(x: fn(&'a u32), y: &'a u32) {} // ok!
+
+struct Foo<'a> { x: &'a u32 }
+
+impl<'a> Foo<'a> {
+    fn bar(&self, x: fn(&'a u32)) {} // ok!
+}
+```
+
+```
+fn bar<'a, F>(x: &F) where F: Fn(&'a u32) {} // ok!
```

Comment:
Why two code blocks here too?

Reply:
Actually there were two examples in the .rs file was thinking that I have to show both codes in the explanation.
But now I have fixed it. Please take a look at it.